cql3/statements/modification_statement.cc

```diff
@@ -416,7 +421,16 @@ modification_statement::execute_without_condition(service::storage_proxy& proxy,
             return now();
         }
 
-        return proxy.mutate_with_triggers(std::move(mutations), cl, timeout, false, qs.get_trace_state(), qs.get_permit(), this->is_raw_counter_shard_write());
+        auto mutate = [&, cl, timeout] (std::vector<mutation> mutations) {
+            return proxy.mutate_with_triggers(std::move(mutations), cl, timeout, false, qs.get_trace_state(), qs.get_permit(), this->is_raw_counter_shard_write());
+        };
+
+        if (qs.get_client_state().is_internal() || qs.get_client_state().is_thrift() || !s->cdc_enabled()) {
+            return mutate(std::move(mutations));
+        }
+        return cdc::apply(proxy, s, timeout, qs.get_permit(), std::move(mutations)).then([cl, timeout, &proxy, &qs, mutate = std::move(mutate)] (std::vector<mutation> mutations) {
```

Comment:
Given that schema is reachable from each mutation, we could just as well move the cdc call into storage proxy (and generalize it to extract the schema set from mutation set -> batch support (?)). 
Not sure how much it adds in simplicity for alternator etc.

Reply:
I think what shows up in the log should depend on the frontend used. For example, each frontend has its own notion of "operation type".